src/display/editor/signature.js

```diff
@@ -98,70 +122,87 @@ class SignatureEditor extends DrawingEditor {
-        }
-        resolve();
-      },
-      { signal }
-    );
-    input.addEventListener("cancel", resolve, { signal });
-    input.click();
-
-    const bitmap = await promise;
-    const {
-      rawDims: { pageWidth, pageHeight },
-      rotation,
-    } = this.parent.viewport;
-    let drawOutlines;
-    if (bitmap?.bitmap) {
-      drawOutlines = SignatureExtractor.process(
-        bitmap.bitmap,
-        pageWidth,
-        pageHeight,
-        rotation,
-        SignatureEditor._INNER_MARGIN
-      );
+  addSignature(outline, heightInPage) {
+    const { x: savedX, y: savedY } = this;
+    this.#isExtracted = outline instanceof ContourDrawOutline;
+    let drawingOptions;
+    if (this.#isExtracted) {
+      drawingOptions = SignatureEditor.getDefaultDrawingOptions();
     } else {
-      drawOutlines = SignatureExtractor.extractContoursFromText(
-        "Hello PDF.js' World !!",
-        { fontStyle: "italic", fontWeight: "400", fontFamily: "cursive" },
-        pageWidth,
-        pageHeight,
-        rotation,
-        SignatureEditor._INNER_MARGIN
-      );
+      drawingOptions = SignatureEditor._defaultDrawnSignatureOptions.clone();
+      drawingOptions.updateProperties({ "stroke-width": outline.thickness });
     }
     this._addOutlines({
-      drawOutlines,
-      drawingOptions: SignatureEditor.getDefaultDrawingOptions(),
+      drawOutlines: outline,
+      drawingOptions,
     });
+    const [parentWidth, parentHeight] = this.parentDimensions;
+    const [, pageHeight] = this.pageDimensions;
+    let newHeight = heightInPage / pageHeight;
+    newHeight = newHeight >= 1 ? 0.5 : newHeight;
+    this.width *= newHeight / this.height;
+    this.height = newHeight;
+    this.setDims(parentWidth * this.width, parentHeight * this.height);
+    this.x = savedX;
+    this.y = savedY;
+    this.center();
+
+    this._onResized();
     this.onScaleChanging();
     this.rotate();
+
     this.div.hidden = false;
   }
+
+  extractSignature(bitmap) {
+    const {
+      rawDims: { pageWidth, pageHeight },
+      rotation,
+    } = this.parent.viewport;
+    return SignatureExtractor.process(
+      bitmap,
+      pageWidth,
+      pageHeight,
+      rotation,
+      SignatureEditor._INNER_MARGIN
+    );
+  }
+
+  extractFromText(text, fontInfo) {
+    const {
+      rawDims: { pageWidth, pageHeight },
+      rotation,
+    } = this.parent.viewport;
+    return SignatureExtractor.extractContoursFromText(
+      text,
+      fontInfo,
+      pageWidth,
+      pageHeight,
+      rotation,
+      SignatureEditor._INNER_MARGIN
+    );
+  }
+
+  getDrawnSignature(curves) {
+    const {
+      rawDims: { pageWidth, pageHeight },
+      rotation,
+    } = this.parent.viewport;
+    return SignatureExtractor.processDrawnLines({
+      lines: curves,
+      pageWidth,
+      pageHeight,
+      rotation,
+      innerMargin: SignatureEditor._INNER_MARGIN,
+      mustSmooth: false,
+      areContours: false,
+    });
+  }
```

Comment:
Could we perhaps use a slightly more consistent naming format for these methods (since a couple uses "extract" and one uses "get")?